TripPlanner/View Controllers/TripViewController.m

```diff
@@ -0,0 +1,259 @@
+                                                            preferredStyle:(UIAlertControllerStyleAlert)];
+    
+    // add the camera action to the alertController
+    [alert addAction:cameraAction];
+    
+    // add the album action to the alert controller
+    [alert addAction:albumAction];
+    
+    return alert;
+    
+}
+
+- (PFFileObject *)getPFFileFromImage: (UIImage * _Nullable)image {
+    
+    // check if image is not nil
+    if (!image) {
+        return nil;
+    }
+    
+    NSData *imageData = UIImagePNGRepresentation(image);
+    // get image data and check if that is not nil
+    if (!imageData) {
+        return nil;
+    }
+    
+    return [PFFileObject fileObjectWithName:@"image.png" data:imageData];
+}
+
+- (BOOL)hasGuest: (NSString *)username guests: (NSMutableArray *)guests {
+    for(NSString *guestUsername in guests) {
+        if ([username isEqualToString:guestUsername]) {
+            return YES;
+        }
+    }
+    return NO;
+}
+
+- (IBAction)onAdd:(id)sender {
+    // check if user is in database
+    NSString *username = self.guestsField.text;
+    PFQuery *query = [PFUser query];
+    [query whereKey:@"username" equalTo:username];
+    
+    [query findObjectsInBackgroundWithBlock:^(NSArray * _Nullable users, NSError * _Nullable error) {
+        // user does not exist
+        if(users.count == 0) {
+            UIAlertController *alert = [self createUserAlert:@"Error Adding Guest" action:@"Cancel" message:@"Username does not exist"];
+            [self presentViewController: alert animated:YES completion:^{
+            }];
+        }
+        // user is already added
+        else if([self hasGuest:username guests:self.guestUsernames]) {
+            UIAlertController *alert = [self createUserAlert:@"Error Adding Guest" action:@"Cancel" message:@"User already added"];
+            [self presentViewController: alert animated:YES completion:^{
+            }];
+        }
+        // user is current user
+        else if([username isEqualToString:PFUser.currentUser.username]) {
+            UIAlertController *alert = [self createUserAlert:@"Error Adding Guest" action:@"Cancel" message:@"Can't add yourself"];
+            [self presentViewController: alert animated:YES completion:^{
+            }];
+        }
+        // user can be added to list
+        else {
+            UIAlertController *alert = [self createUserAlert:@"Successful" action:@"Ok" message:@"Guest was successfully added"];
+            [self presentViewController: alert animated:YES completion:^{
+            }];
+            self.guestList.text = [NSString stringWithFormat:@"%@%@, ", self.guestList.text, username];
+            [self.guestUsernames addObject:username];
+            [self.guests addObject:users[0]];
+        }
+        self.guestsField.text = @"";
+    }];
+}
+
+- (UIAlertController *)createUserAlert: (NSString *)title action:(NSString *)action message:(NSString *)message{
```

Comment:
this looks like a common-enough function that it can be in its own, separate class. Other controllers may want to use it, and they shouldn't have to reference this view controller to do so.